api/rest/restapi.go

```diff
@@ -585,6 +585,69 @@ func (api *API) allocationHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// defined groups / aliases for filtering
+var filterGroups = map[string][]string{
+	"error":  {"error", "pin_error", "unpin_error", "cluster_error"},
+	"queued": {"queued", "pin_queued", "unpin_queued"},
+}
+
+// check if any of the strings in matches are in the list
+func anyStringInSlice(matches, list []string) bool {
+	for _, b := range list {
+		for _, d := range matches {
+			if len(b) > 0 && d == b {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func checkFilterMatches(filterList []string, value string) bool {
+	// in case we have not defined any alises for the passed string, just use the string itself
+	if alias, exists := filterGroups[value]; exists {
+		return anyStringInSlice(alias, filterList)
+	}
+	return anyStringInSlice([]string{value}, filterList)
+}
+
+func filterStatusAll(filter string, pinInfos []types.GlobalPinInfoSerial) []types.GlobalPinInfoSerial {
+	if filter == "" {
+		return pinInfos
+	}
+	filterList := strings.Split(strings.ToLower(filter), ",")
+	for i := len(pinInfos) - 1; i >= 0; i-- {
```

Comment:
This would be one of those loops where a [`label`](https://golang.org/ref/spec#Label_scopes) really simplifies the code:
```go
	var filtered []types.GlobalPinInfoSerial

NEXTPI:
	for _, pi := range pinInfos {
		for _, entry := range pi.PeerMap {
			if anyStringInSlice(filterList, entry.Status) {
				filtered = append(filtered, pi)
				continue NEXTPI
			}
		}
	}
```

Reply:
So in this case, it is preferred in general, instead of removing from the existing slice, to build up a new slice instead? I will apply this.

Reply:
@lanzafame If you have a chance, can you link to a doc or relay how to start the mock RPC and run the tests in restapi_test.go ? Thank you!

Reply:
@sona1111 to run the api tests just do `go test -v ./api/rest` (from the repo root)

Reply:
(after `make deps`, that is)